GitCommands/Git/GitModule.cs

```diff
@@ -3092,17 +3101,7 @@ public IReadOnlyList<string> GetMergedRemoteBranches()
 
         public IReadOnlyList<IGitRef> ParseRefs(string refList)
         {
-            // Parse lines of format:
-            //
-            // 69a7c7a40230346778e7eebed809773a6bc45268 refs/heads/master
-            // 69a7c7a40230346778e7eebed809773a6bc45268 refs/remotes/origin/master
-            // 366dfba1abf6cb98d2934455713f3d190df2ba34 refs/tags/2.51
-            //
-            // Lines may also use \t as a column delimiter, such as output of "ls-remote --heads origin".
-
-            Regex regex = new(@"^(?<objectid>[0-9a-f]{40})[ \t](?<refname>.+)$", RegexOptions.Multiline | RegexOptions.Compiled);
```

Comment:
Lines that start with sha should use TryParse() and trim whitespace for the rest, likely faster.
Maybe a separate PR.